crates/model/src/order.rs

```diff
@@ -644,7 +646,21 @@ pub enum OrderClass {
     /// the future, then fulfill my order". These orders have their fee set to zero, because it's
     /// impossible to predict fees that far in the future. Instead, the fee is taken from the order
     /// surplus once the order becomes fulfillable and the surplus is high enough.
-    Limit,
+    Limit(LimitOrderClass),
+}
+
+impl OrderClass {
+    pub fn is_limit(&self) -> bool {
+        matches!(self, Self::Limit(_))
+    }
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Default, Hash, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LimitOrderClass {
```

Comment:
naming nit: The name `LimitOrderClass` only makes sense if you know that this is supposed to be contained in an `OrderClass::Limit`. Without that context `LimitOrderData` or `SurplusFeeData` could make more sense.
I don't have a strong opinion on this because the limit order namespace is already quite crowded so the ideas I had don't satisfy me either.